dask_expr/_expr.py

```diff
@@ -3190,21 +3187,22 @@ def determine_column_projection(expr, parent, dependents, additional_columns=Non
         column_union = parent.columns.copy()
     parents = [x() for x in dependents[expr._name] if x() is not None]
 
+    seen = set()
     for p in parents:
-        if len(p.columns) > 0:
-            column_union.append(p._projection_columns)
+        if p._name in seen:
+            continue
+        seen.add(p._name)
+
+        column_union.extend(p._projection_columns)
 
     if additional_columns is not None:
-        column_union.append(additional_columns)
+        column_union.extend(flatten(additional_columns, container=list))
 
     # We can end up with MultiIndex columns from groupby ops, needs to be
     # accounted for in the sort
-    flattened_columns = set(flatten(column_union, container=list))
+    flattened_columns = set(column_union)
     try:
-        column_union = sorted(
-            flattened_columns,
-            key=lambda x: x[0] if isinstance(x, tuple) else x or MinType(),
-        )
+        column_union = sorted(flattened_columns)
```

Comment:
Did something change here that I'm not seeing, or was the `key=` never necessary?

Reply:
The logic changed a little bit, tuples now end up in the except branch. Since tuples are rare, it’s better to slow them down than to slow the default path down with the check